### PR TITLE
Issue279 redesign doc

### PR DESCRIPTION
**Pull request by juanbrusco**
- new documentation files & structure
- github action to create python env, install dependencies, generate sphinx-doc & deploy into new branch named "gh-pages"
- updated readme links
- added new sphinx dependencies to setup.py & put all "doc" reqs together

### CODE REVIEW

**Comment by juanbrusco** [comment]
@avisil  @jdpsen  @bhavani105 @franzmpub 
Hi, team.
This PR is ready to be merged.
Could you please, take a look and approve, so we can test documentation live version with all changes

**Review by bhavani105 — COMMENTED** [review]
@franzmpub and i took a quick look at the README under primeqa/ir and found a couple of issues:
1. It is rendering the sphinx template 
2. The hierarchy appears to be incorrect. For example Indexing and Corpus Format are shown as being at the same level. Corpus Format should be below Indexing as in the original README.md

**Comment by juanbrusco** [comment]
@bhavani105  @franzmpub 

1. now the README files are using a part of a Sphinx code, this is necessary to keep the same between packages and docs when the file is updated, and that's why I've added a comment in raw mode
 
image

2.  The hierarchy is ok because in the Sphinx config.py file you define the source and then it takes from there the levels. If that would be wrong, then the api reference won't work.
 
image

**Comment by franzmpub** [comment]
> @bhavani105 @franzmpub
> 
> 1. now the README files are using a part of a Sphinx code, this is necessary to keep the same between packages and docs when the file is updated, and that's why I've added a comment in raw mode
> 
> image
> 
> 2. The hierarchy is ok because in the Sphinx config.py file you define the source and then it takes from there the levels. If that would be wrong, then the api reference won't work.
> 
> image

The point is that the top level README in IR shows now as
image

and the hierarchy of headings is flattened, e.g.

image

as opposed to 

image

and

image

which 1. looks weird 2. reduces readability.

If there is no way to deal with it, we will have to live with it, but it's a substantial compromise / cost paid for compatibility with Sphinx.

**Comment by juanbrusco** [comment]
@franzmpub
Yes, Martin. Got your point ...
I was talking with @bhavani105  about this and implementing a fix to keep the hierarchy. 

Sorry to both for the misunderstood

File: docs/development.md [hunk 1]
```diff
@@ -109,18 +110,20 @@ cd docs
...
 Each contribution should have an easy to use notebook that exemplifiess how to instantiate and use the functionality. 
-An example for MRC on Tydi is here: https://github.ibm.com/ai-foundation/PrimeQA/blob/master/notebooks/mrc/tydiqa.ipynb
+An example for MRC on Tydi is [here](https://github.com/primeqa/primeqa/blob/main/notebooks/mrc/tydiqa.ipynb)
 
-#### Documentation Pages
+#### Documentation Pages:
 Documentation pages (such as this one) are generated from a collection of Markdown files located
-in `docs`. It also contains a subfolder calles `img` which contains any (legally usable) images needed for documentation purposes. 
+in `docs`. It also contains a subfolder calles `_static/img` which contains any (legally usable) images needed for documentation purposes. 
+
```
Comment (bhavani105) [inline docs/development.md:128]: Could we document the requirement to have the sphinx related block in the README.md and what to do when a new README.md is added.
Is there an error message or warning to check if the block is missing.

Comment (juanbrusco) [reply docs/development.md:128]: - I've added a documentation which explains what to do when a new pkg/readme is added, you can check here: https://github.com/primeqa/primeqa/tree/issue279-redesign-doc/docs#new-package-added-to-primeqa & a link here in development.md file -> https://github.com/primeqa/primeqa/blob/issue279-redesign-doc/docs/development.md#deploying-sphinx-docs-as-project-online
- if the block is missing, an warning/error msg will appear on github-action log

**Comment by bhavani105** [comment]
@juanbrusco Please comment on https://github.com/primeqa/primeqa/issues/179
Is this already covered

**Comment by juanbrusco** [comment]
> @juanbrusco Please comment on #179 Do we plan to check/show coverage https://www.sphinx-doc.org/en/master/usage/extensions/coverage.html

@bhavani105 that extension is not set on Sphinx configuration. I'll work on it. 
But, please, could you approve this PR to first of all check that everything is working?
Then I can continue and add the Coverage ext.

**Review by bhavani105 — APPROVED** [review]
LGTM

Checked the READMEs and the development.md

**Review by avisil — APPROVED** [review]

**Review by franzmpub — APPROVED** [review]
LGTM

**Pull request merged**